lambdas/jwtverification/src/main/java/uk/gov/di/ipv/cri/passport/jwtverification/JwtVerificationHandler.java

```diff
@@ -53,22 +60,33 @@ public APIGatewayProxyResponseEvent handleRequest(APIGatewayProxyRequestEvent in
         }
 
         try {
+            SignedJWT signedJWT = SignedJWT.parse(input.getBody());
             Certificate clientCert = configurationService.getClientCert(clientId);
 
-        } catch (CertificateException e) {
-            LOGGER.log(Level.WARNING, "Failed to retrieve certificate");
-            return ApiGatewayResponseGenerator.proxyJsonResponse(BAD_REQUEST, ErrorResponse.FAILED_TO_RETRIEVE_CERTIFICATE);
-        }
+            if (isInvalidSignature(SignedJWT.parse(input.getBody()), clientCert)) {
```

Comment:
You don't need to parse the JWT again here, instead just pass in the signedJwt from above

```suggestion
            if (isInvalidSignature(signedJwt, clientCert)) {
```

Reply:
Refactored